### DISCUSSION TITLE
Leverage Cern Root

### DISCUSSION THREAD

**Discussion by alejandrogallo** [Lisp-Stat/lisp-stat#19] [Ideas]
I have asked myself always why people don't leverage more Root https://root.cern/.
It seems that providing high quality bindings from common lisp would be super useful,
both for statistics, plotting and even for 3d graphics https://root.cern.ch/doc/master/group__opengl.html

Has anyone tried to really consider this?

**Comment by snunez1** [comment]
That's an interesting idea, and bindings would be useful.  There's [cephes.cl](https://github.com/Lisp-Stat/cephes.cl) for many statistical functions, but Root seems to cover a lot more.

I think a wrapper is possible to build now, but would be 'cutting edge'.  The reason is that wrapping C++ has historically been a challenge in Common Lisp.  Wrapping C is is easy.  There are a couple of systems for wrapping C++ that have been released in the last year that could be used.  I suspect something this large will expose some bugs, so whoever does this will have to work with the system authors (of the C++ wrapping systems) to resolve any issues.

I hope someone picks this idea up.

**Reply by alejandrogallo** [reply]
cephes.cl is great.
Maybe next year if I still want to use it I'll give it a try.
The thing that makes me a bit
optimistic and pesimistic is that root is completely interfaced with
python.

  https://github.com/root-project/root/tree/master/bindings

I'm not an expert in bindings but that looks like a lot of work.
However I think even if daunting, it would supercharge the common lisp ecosystem.

**Reply by snunez1** [reply]
If I were doing this, I'd use an autowrap.  Cephes was small enough to do by hand.  There are some large-ish wrappers of C libraries done with auto-wrap.  It _should_ also work with C++, but it's untested as far as I know.
